sender_test.cpp

```diff
@@ -0,0 +1,32 @@
+#define CATCH_CONFIG_MAIN
+#include "test/catch.hpp"
+#include <iostream>
+#include "sensor/temperature.h"
+#include "sensor/state_of_charge.h"
+#include <vector>
+#include "sender/sender.h"
+
+using namespace std;
+
+TEST_CASE("Test 1 : Check whether 50 random temperature values are generated or not") {
+  vector<int> generatedTemperatureValues = generateTemperatureValues(TEMPERATURE_MAX_VALUE ,TEMPERATURE_MIN_VALUE, READINGS_IN_STREAM);
+  bool isTemperatureBufferEmpty = generatedTemperatureValues.empty();
+  REQUIRE(isTemperatureBufferEmpty == false);
+  int temperatureBufferSize = generatedTemperatureValues.size();
+  REQUIRE(temperatureBufferSize == READINGS_IN_STREAM);
+}
+
+TEST_CASE("Test 2 : Check whether 50 random SOC values are generated or not") {
+  vector<int> generatedSocValues = generateSocValues(SOC_MAX_VALUE ,SOC_MIN_VALUE, READINGS_IN_STREAM);
+  bool isSocBufferEmpty = generatedSocValues.empty();
+  REQUIRE(isSocBufferEmpty == false);
+  int socBufferSize = generatedSocValues.size();
+  REQUIRE(socBufferSize == READINGS_IN_STREAM);
+}

```

Comment:
I feel still some more test case could be added 
1. by passing READING_IN_STREAM to 0.
2. Passing both SOC_MAX and SOC_MIN to same constant.

Reply:
changes done.